src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/service/ReportService.kt

```diff
@@ -31,6 +35,13 @@ class ReportService(
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
+  fun getReports(
+    pageable: Pageable = PageRequest.of(0, 20, Sort.by("incidentDateAndTime").descending()),
```

Comment:
Is date the best sort or the Incident report ID? I think we want a consistent list order.  i.e. the reports come out in the same order each time.  Could use the created date instead also - its just that incident Date /Time could have a new report added in the middle

Reply:
this is just the default and sort orders can be specified by the client. i figured that this endpoint will in future be used to answer “gimme the latest reports in MDI” so the date the incident occurred seemed a reasonable default.

the reconciliation process should probably not use a date as it won't result in a stable order (they’re non-unique columns). actually, especially so with created date – we don’t migrate that info from NOMIS so it’s the moment of migration that gets stored. i think that reconciliation should sort on incident number (which works according to the test case) or id.

Reply:
I think for the purpose of reconciliation return them sorted by the ID may be the best thing to do. It should be fast as it's the PK and it avoid the scenario where new reports may be added to older pages.

Reply:
i updated my comment a bit late but we cannot sort the reports by id and have the order be stable since _these_ uuids are essentially random.

still, it’s up to the client to decide what field to order by 🤷

Reply:
Oh they're UUID, so that's true! In that case sorting by date is more sensible.

Maybe sort by creation date would be better and there is less chance of newer reports getting in the "middle"?

Reply:
>Maybe sort by creation date would be better and there is less chance of newer reports getting in the "middle"?

creation date is still a problem for reports that come from nomis:

>the reconciliation process should probably not use a date as it won't result in a stable order (they’re non-unique columns). actually, especially so with created date – we don’t migrate that info from NOMIS so it’s the moment of migration that gets stored.

Reply:
Yes don't use UUID for sorting. But we have the incident reporting number which is unique